tez-dag/src/main/java/org/apache/tez/dag/app/dag/impl/TaskAttemptImpl.java

```diff
@@ -228,6 +228,7 @@ public static DataEventDependencyInfo fromProto(DataEventDependencyInfoProto pro
   private final boolean leafVertex;
   
   private TezTaskAttemptID creationCausalTA;
+  private Set<NodeId> unhealthyNodesHistory;
```

Comment:
From a design perspective, I wonder if the task attempt owns the list of nodes to avoid. Really this data structure is owned by the Task and used by the Task Attempt. Unfortunately, it seems TaskAttemptImpl doesn't have its own Task like getTask(). Then we could move the logic of avoidance to TaskImpl. This is probably acceptable as is, but it points to needing some refactoring of this class in the future. So I don't think a change should be made as part of this jira.

Reply:
Agreed. Were in the same shoes when making this change. Will open a JIRA to do this separately.

Reply:
TEZ-4045 has been created to track this. @Chyler @jteagles